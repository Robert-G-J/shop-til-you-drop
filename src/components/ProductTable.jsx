import React from "react";
import PropTypes from "prop-types";
import ProductTableRow from "./ProductTableRow";

const ProductTable = ({ updateQuantity, products, basket }) => (
  <table className="product-table">
    <thead>
      <tr className="table-row header">
        <th className="table-cell header" />
        <th className="table-cell header">Price</th>
        <th className="table-cell header quantity">Quantity</th>
        <th className="table-cell header total">Total</th>
      </tr>
    </thead>
    <tbody>
      {products.map(product => (
        <ProductTableRow
          key={`${product.id}+${Date.now()}`}
          id={product.id}
          name={product.name}
          description={product.description}
          price={product.price}
          updateQuantity={updateQuantity}
        />
      ))}
    </tbody>
  </table>
);

ProductTable.propTypes = {
  products: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number.isRequired,
      name: PropTypes.string.isRequired,
      description: PropTypes.string.isRequired,
      price: PropTypes.string.isRequired,
      updateQuantity: PropTypes.func.isRequired
    })
  ),
  basket: PropTypes.arrayOf(PropTypes.shape({}))
};
export default ProductTable;
