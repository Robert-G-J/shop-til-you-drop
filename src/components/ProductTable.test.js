import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import ProductTable from "./ProductTable";
import productData from "../data/unitPriceList.json";

describe("<ProductTable />", () => {
  const props = productData;

  it("renders correctly", () => {
    const tree = renderer.create(<ProductTable {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders without crashing", () => {
    const component = shallow(<ProductTable {...props} />);
    expect(component.exists()).toEqual(true);
  });
});
