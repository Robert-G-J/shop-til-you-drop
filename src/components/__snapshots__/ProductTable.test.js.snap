// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductTable /> renders correctly 1`] = `
<table
  className="product-table"
>
  <thead>
    <tr
      className="table-row header"
    >
      <th
        className="table-cell header"
      />
      <th
        className="table-cell header"
      >
        Price
      </th>
      <th
        className="table-cell header quantity"
      >
        Quantity
      </th>
      <th
        className="table-cell header total"
      >
        Total
      </th>
    </tr>
  </thead>
  <tbody>
    <tr
      className="table-row"
    >
      <td
        className="table-cell id"
      >
        1
      </td>
      <td
        className="table-cell product"
      >
        plumbus
        <p
          className="description"
        >
          A Plumbus will aid many things in life, making life easier. With proper maintenance, handling, storage, and urging, Plumbus will provide you with a lifetime of better living and happiness
        </p>
      </td>
      <td
        className="table-cell price "
      >
        £
        399.99
      </td>
      <td
        className="table-cell quantity"
      >
        <input
          className="quantity"
          defaultValue="1"
          min="1"
          name="quantity"
          type="number"
        />
      </td>
      <td
        className="table-cell total"
      >
        0
      </td>
    </tr>
    <tr
      className="table-row"
    >
      <td
        className="table-cell id"
      >
        2
      </td>
      <td
        className="table-cell product"
      >
        butter robot
        <p
          className="description"
        >
          Small, two armed, mobile robot created for the sole purpose of passing butter
        </p>
      </td>
      <td
        className="table-cell price "
      >
        £
        49.99
      </td>
      <td
        className="table-cell quantity"
      >
        <input
          className="quantity"
          defaultValue="1"
          min="1"
          name="quantity"
          type="number"
        />
      </td>
      <td
        className="table-cell total"
      >
        0
      </td>
    </tr>
    <tr
      className="table-row"
    >
      <td
        className="table-cell id"
      >
        3
      </td>
      <td
        className="table-cell product"
      >
        szechuan sauce
        <p
          className="description"
        >
          Szechuan Chicken McNugget Sauce, a dipping sauce sachet for McDonald's Chicken McNuggets. AKA Mulan McNugget Sauce
        </p>
      </td>
      <td
        className="table-cell price "
      >
        £
        1.30
      </td>
      <td
        className="table-cell quantity"
      >
        <input
          className="quantity"
          defaultValue="1"
          min="1"
          name="quantity"
          type="number"
        />
      </td>
      <td
        className="table-cell total"
      >
        0
      </td>
    </tr>
    <tr
      className="table-row"
    >
      <td
        className="table-cell id"
      >
        4
      </td>
      <td
        className="table-cell product"
      >
        microverse battery
        <p
          className="description"
        >
          Contains a miniature universe with a planet inhabited by intelligent life. These lifeforms use kinetic devices to produce electricity. Under the guise of "waste power" a majority of the energy produced is extracted to power a small personal spaceship
        </p>
      </td>
      <td
        className="table-cell price "
      >
        £
        90.00
      </td>
      <td
        className="table-cell quantity"
      >
        <input
          className="quantity"
          defaultValue="1"
          min="1"
          name="quantity"
          type="number"
        />
      </td>
      <td
        className="table-cell total"
      >
        0
      </td>
    </tr>
  </tbody>
</table>
`;
